File: Javascript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var lookup = {};
    var res = [];
    nums.some((v, i) => {
        if (lookup[target - v]) {
            res = [lookup[target - v], i];
            return true;
        } else {
            lookup[v] = i;
            return false;
        }
    });
    return res;
};

File: Golang/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	lookup := make(map[int]int)
	for i, num := range nums {
		if j, ok := lookup[target-num]; ok {
			return []int{j, i}
		}
		lookup[nums[i]] = i
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{2,7,11,15}, 9))
}
